dipy/denoise/fast_lpca.pyx

```diff
@@ -0,0 +1,307 @@
+                            # q=(0, 1, 1)
+                            update_vector(mu, S[cur_i, j - 1, k - 1], -1)
+                    if k > 0:
+                        # q=(0, 0, 1)
+                        update_matrix(cov, T[cur_i, j, k - 1], 1)
+                        update_vector(mu, S[cur_i, j, k - 1], 1)  # q=(0, 0, 1)
+
+                    # Add displaced signed corners
+                    if i >= m:
+                        x = I[i - m, j, k]  # q=(1, 0, 0)
+                        update_outer_prod(cov, x, -1)
+                        update_vector(mu, x, -1)
+                        if j >= m:
+                            x = I[i - m, j - m, k]  # q=(1, 1, 0)
+                            update_outer_prod(cov, x, 1)
+                            update_vector(mu, x, 1)
+                            if k >= m:
+                                x = I[i - m, j - m, k - m]  # q=(1, 1, 1)
+                                update_outer_prod(cov, x, -1)
+                                update_vector(mu, x, -1)
+                        if k >= m:
+                            x = I[i - m, j, k - m]  # q=(1, 0, 1)
+                            update_outer_prod(cov, x, 1)
+                            update_vector(mu, x, 1)
+                    if j >= m:
+                        x = I[i, j - m, k]  # q=(0, 1, 0)
+                        update_outer_prod(cov, x, -1)
+                        update_vector(mu, x, -1)
+                        if k >= m:
+                            x = I[i, j - m, k - m]  # q=(0, 1, 1)
+                            update_outer_prod(cov, x, 1)
+                            update_vector(mu, x, 1)
+                    if k >= m:
+                        x = I[i, j, k - m]  # q=(0, 0, 1)
+                        update_outer_prod(cov, x, -1)
+                        update_vector(mu, x, -1)
+
+                    # Save current integrals for future reference
+                    update_matrix(T[cur_i, j, k], cov, 0)
+                    update_vector(S[cur_i, j, k], mu, 0)
+
+                    # Use integral of current rectangle
+                    # Last corner is (i, j, k) => center is (i-radius,
+                    # j-radius, k-radius)
+                    if (i >= m - 1) and (j >= m - 1) and (k >= m - 1):
+                        # Divide by the number of samples
+                        for ii in range(ndiff):
+                            mu[ii] /= nsamples
+                            for jj in range(ndiff):
+                                cov[ii, jj] /= nsamples
+                        # Subtract self-outer product of mu from cov
+                        update_outer_prod(cov, mu, -1)
+
+                        i0 = i - radius
+                        j0 = j - radius
+                        k0 = k - radius
+
+                        with gil:
+                            # eigen value decomposition
+                            d, W_hat = np.linalg.eigh(cov)
+
+                        l0norm = 0
```

Comment:
@omarocegueda yeah that makes sense :), it improved the performance a bit! I verified this and it works!